applications/portal/frontend/src/components/home/ExplorationSpinalCordDialog.tsx

```diff
@@ -36,35 +27,106 @@ const useStyles = makeStyles(() => ({
       marginBottom: '0.75rem'
     },
   },
+  editorWrapper: {
+    '&:hover': {
+      borderColor: '#fff'
+    },
+    '&:focus-within': {
+      borderColor: '#7b61ff'
+    },
+  },
 }));
 
+const DESCRIPTION_KEY = "description";
+
 export const ExplorationSpinalCordDialog = (props: any) => {
+
+  const api = WorkspaceService.getApi();
   const classes = useStyles();
-  const { open, handleClose } = props;
+  const commonClasses = common();
+  const { open, handleClose, experiment, tagsOptions, onExperimentChange } = props;
+
+  const [name, setName] = React.useState<string>(experiment.name);
+  const [description, setDescription] = React.useState<string>(experiment.description);
+  const [initialTags, setInitialTags] = React.useState(experiment.tags.map((t: any) => t.name));
+  const [tags, setTags] = React.useState(experiment.tags.map((t: any) => t.name));
+  const [validationErrors, setValidationErrors] = React.useState(new Set([]));
+
+  const handleFormChange = (newValue: any, setState: any, errorKey: string) => {
+    validationErrors.delete(errorKey);
+    setValidationErrors(validationErrors);
+    setState(newValue);
+  };
+
+  const onTagsChange = (e: any, value: string[]) => {
+    setTags(value);
+  };
+
+  const deleteTag = async (tag: string) => {
+    try {
+      await api.deleteTagExperiment(experiment.id.toString(), tag);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const addTags = async (tags: string[]) => {
+    try {
+      await api.addTagsExperiment(experiment.id.toString(), tags);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  const handleAction = async () => {
+    const deletedTags = initialTags.filter((tag: any) => !tags.includes(tag));
```

Comment:
The action should only occur if there are no errors in the form. Aka, all the required fields are filled.
Name and description are required fields